mod/badges/set/type_plus_right/metric_answer/value/award_affinity_badges.rb

```diff
@@ -41,7 +40,8 @@ def affinity_name affinity, project_card=nil
 end
 
 def create_count restriction={}
-  Answer.where(restriction.merge(creator_id: Auth.current_id)).count
+  Answer.where(restriction.merge(creator_id: Auth.current_id))
+        .where.not(answer_id: nil).count
```

Comment:
this means virtual answers will no longer contribute to awards.  (Though, in fact, they may have often _prevented_ awards, because when multiple items are created in a single act, I think it's possible to skip/miss a threshold)